src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientProduceFactory.java

```diff
@@ -636,8 +642,7 @@ private void flushClientFanInitialIfNecessary(
 
                 final KafkaCacheEntryFW nextEntry = cursor.next(entryRO);
 
-                if (nextEntry != null &&
-                    (nextEntry.flags() & CACHE_ENTRY_FLAGS_COMPLETED) == CACHE_ENTRY_FLAGS_COMPLETED)
+                if (nextEntry != null)
```

Comment:
This will change behavior and advance before entry is completed, even for non-dirty case, right?

Suggest creating a new constant
```
private static final int CACHE_ENTRY_FLAGS_ADVANCE = CACHE_ENTRY_FLAGS_COMPLETED | CACHE_ENTRY_FLAGS_DIRTY;
```
then changing previous condition to the following
```
if (nextEntry != null &&
    (nextEntry.flags() & CACHE_ENTRY_FLAGS_ADVANCE) != 0)
```
so that it advances if `completed` or `dirty` but not if neither flag is set.